wai-aria/scripts/aria-utils.js

```diff
@@ -156,6 +156,21 @@ const AriaUtils = {
     }
   },
 
-
+  verifyLevelsBySelector: function (selector) {
```

Comment:
Sorry Adam, I should have clarified that we can't test `level` yet. There's no WebDriver accessor for `level` or similar properties beyond role and label. This checks the IDL, but that will not test the accessibility internals we need.

To clarify, the WPT I thought you were going to write would only be able to check what roles were applied, if any, when HTML and ARIA headings are combined/nested in this unexpected way. From the AT behavior, it seems like all engines are doing different things with this particular bit of markup.

Using the original example:

```html
<div role=heading aria-level=3>
  <a role=heading aria-level=3>
    <h2>What Sports Illustrated's AI scandal...</h2>
  </a>
</div>
```

Do we get multiple headings in the browser internals? Or 1? Or something else? What about other similar variants. We could do a phone call if you like.

Reply:
Understood, thanks. A quick call would be great — I’ll DM you. 🙏🏻

Reply:
There might also be some way to test if unexpected heading nesting affects label, such as:

```html
<div role=heading aria-level=1>
    one
    <h2>two</h2>
</div>
```

e.g. If this results in two headings, is it because of nested containment (first heading label may be "one two") or element continuation (first heading label may be "one") 

But I doubt there would be any detectable difference if you changed the level numbers.